### PR TITLE
Implement isnumeric for Str and write test case

### PR DESCRIPTION
**Pull request by atmishra**
Refs #36 This is my first PR. I have tried implementing isnumeric method for Str and have written some test case.

### CODE REVIEW

File: python/common/org/python/types/Str.java [hunk 1]
```diff
@@ -860,9 +860,18 @@ private long toPositiveIndex(long index) {
...
+        }
+        for (char ch : this.value.toCharArray()) {
+            if (!(Character.isDigit(ch))) {
+                return new org.python.types.Bool(false);
+            }
+        }
+        return new org.python.types.Bool(true);
     }
 
+
```
Comment (cflee) [inline python/common/org/python/types/Str.java]: I believe there's just one newline between the other methods in this class – so I think it's not necessary to add this newline here.

File: tests/datatypes/test_str.py [hunk 2]
```diff
@@ -681,6 +681,12 @@ def __str__(self):
                 print(err)
             """)
 
+    def test_isnumeric(self):
+        self.assertCodeExecution("""
+        for str_ in ['123', '123.4', 'abc', '', ' ', 'ABCD', 'ABCD ', '12323445']:
```
Comment (cflee) [inline tests/datatypes/test_str.py]: Could you also test some other cases like `'123.', '.12', '1A', 'A1'` to make sure the edge cases are handled correctly?

**Comment by atmishra** [comment]
Did the requested changes.

**Comment by cflee** [comment]
Looks great, thanks for the contribution! ✨
